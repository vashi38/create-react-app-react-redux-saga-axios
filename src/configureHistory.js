/* eslint-disable react-hooks/rules-of-hooks */

import { browserHistory, useRouterHistory } from 'react-router';
import { createHistory } from 'history';

function configureDevHistory() {
    return useRouterHistory(createHistory)({
        basename: '/',
    });
}

function configureProdHistory() {
    return browserHistory;
}

function configureTestingHistory() {
    return useRouterHistory(createHistory)({
        basename: '/',
    });
}

export function configureHistory() {
    const env = process.env.NODE_ENV;
    switch (env) {
    case 'development':
        return configureDevHistory();
    case 'testing':
        return configureTestingHistory();
    default:
        return configureProdHistory();
    }
}
