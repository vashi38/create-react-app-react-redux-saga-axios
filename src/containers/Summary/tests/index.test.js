import React from 'react';
import { render } from '@testing-library/react';

import { Provider } from 'react-redux';
import { configureHistory } from '../../../configureHistory';
import configureStore from '../../../utilities/store';

import HomePage from '../../Home';
import SummaryPage from '../';

const browserHistory = configureHistory();

describe('<summary>', () => {
    let store;
    beforeAll(() => {
        store = configureStore({}, browserHistory);
    })
    it('should render and match the snapshot', () => {
        const {
          container: { firstChild },
        } = render(
          <Provider store={store}>
              <HomePage 
                children={SummaryPage}
              />
          </Provider>,
        );
        expect(firstChild).toMatchSnapshot();
    });
})