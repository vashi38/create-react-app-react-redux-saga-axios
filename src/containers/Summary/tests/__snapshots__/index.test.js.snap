// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<summary> should render and match the snapshot 1`] = `
<div
  class="container"
  data-reactroot=""
>
  <div
    class="appHeaderContainer"
  >
    <div
      class="breadcrumbContainer"
    />
    <a>
      Total Revenue
    </a>
  </div>
  <div
    class="routeComponentContainer"
  />
</div>
`;
